### PR TITLE
Finish up with user endpoints

### PR DESCRIPTION
**Pull request by GitaGliaudyte**
Added more response codes to user endpoints. Also, I've tried to organize front files a bit and added login form. For now, I use localStorage to store a token, although I'm not sure if this is needed and wanted.

### CODE REVIEW

File: OMGVA PoS/Business layer/Controllers/UserController.cs [hunk 1]
```diff
@@ -48,43 +58,117 @@ public async Task<ActionResult<LoginDTO>> Login([FromBody]LoginRequest loginRequ
...
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status403Forbidden)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+        public IActionResult Update([FromBody] UpdateUserRequest user, long id)
         {
-            _userRepository.UpdateUser(id, user);
-            if(_userRepository.GetUser(id).Equals(user))
-                return Ok();
+            JwtSecurityToken token = JwtTokenHelper.GetJwtToken(HttpContext.Request.Headers.Authorization);
```
Comment (AugustinasJarockis) [inline OMGVA PoS/Business layer/Controllers/UserController.cs:110]: You should check somewhere if user email is valid.

Comment (AugustinasJarockis) [reply OMGVA PoS/Business layer/Controllers/UserController.cs:110]: There is an extension for that.

**Review by AugustinasJarockis — CHANGES_REQUESTED** [review]

File: OMGVA PoS/Business layer/Controllers/UserController.cs [hunk 2]
```diff
@@ -48,43 +58,117 @@ public async Task<ActionResult<LoginDTO>> Login([FromBody]LoginRequest loginRequ
...
+                if (token.UserRoleEquals(UserRole.Owner) && !token.UserBusinessEquals((long)user.BusinessId))
+                    return Forbid();
+
+                return Ok(JsonConvert.SerializeObject(user));
+            }
+            catch (KeyNotFoundException ex)
+            {
+                return NotFound(ex.Message);
+            }
+            catch (Exception ex)
```
[the remark this answers is not in the document]
Comment (GitaGliaudyte) [reply OMGVA PoS/Business layer/Controllers/UserController.cs:95]: I thought of printing exceptions with logger, therefore it's like a placeholder

[the remark this answers is not in the document]
Comment (AugustinasJarockis) [reply OMGVA PoS/Business layer/Controllers/UserController.cs:95]: Ah, ok then.

File: OMGVA PoS/Business layer/Controllers/UserController.cs (hunk 1 above)
Comment (AugustinasJarockis) [reply OMGVA PoS/Business layer/Controllers/UserController.cs:128]: *String extension method

File: OMGVA PoS/Business layer/Controllers/UserController.cs [hunk 3]
```diff
@@ -116,6 +134,9 @@ public IActionResult Update([FromBody] UpdateUserRequest user, long id)
 
             try
             {
+                if(user.Email != null && !user.Email.IsValidEmail())
```
Comment (AugustinasJarockis) [inline OMGVA PoS/Business layer/Controllers/UserController.cs:137]: IsValidEmail checks if the string is null therefore the first check is unnecessary.

**Review by AugustinasJarockis — APPROVED** [review]
Well done. One negligible defect still remains but it barely matters.

**Review by AugustinasJarockis — APPROVED** [review]
Great!

**Pull request merged**